internal/services/postgres/postgresql_flexible_server_resource.go

```diff
@@ -382,7 +382,6 @@ func resourcePostgresqlFlexibleServer() *pluginsdk.Resource {
 			}
 
 			newMb = newStorageMbRaw.(int)
-			newTier = newTierRaw.(string)
```

Comment:
I assume that the root cause is TF always gets the value in the previous apply when `storage_tier` isn't set in the tf config once `Computed: true` is added in the schema. So we may need to use `IsExplicitlyNullInConfig` to check if it's set in the tf config . Below is the sample for the fix.

```
oldTierRaw, newTierRaw := diff.GetChange("storage_tier")

// storage_tier is added `Computed: true` so that TF always gets the value in the previous apply when `storage_tier` isn't set in the tf config. So we need to use `IsExplicitlyNullInConfig` to check if it's set in the tf config
			if v := diff.GetRawConfig().AsValueMap()["storage_tier"]; v.IsNull() {
				newTierRaw = ""
			}
```